iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/orgalternativenames/CsvOrganizationAltNamesDictionaryFactory.java

```diff
@@ -0,0 +1,60 @@
+package eu.dnetlib.iis.wf.affmatching.orgalternativenames;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+/**
+ * Factory of alternative names dictionary. As source of dictionary
+ * it uses csv files located in the classpath.
+ * 
+ * @author madryk
+ */
+public class CsvOrganizationAltNamesDictionaryFactory {
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Creates dictionary of alternative organization names using csv files located 
+     * under provided classpaths.
+     */
+    public List<List<String>> createAlternativeNamesDictionary(List<String> classpathCsvResources) throws IOException {
+        
+        List<List<String>> dictionary = Lists.newArrayList();
```

Comment:
Have you considered using `List` of `Set` of `String`s? This will eliminate potential duplicates right away.